Sources/Datadog/RUM/RUMEvent/RUMEventEncoder.swift

```diff
@@ -15,6 +15,9 @@ internal struct RUMEvent<DM: RUMDataModel>: Encodable {
     let attributes: [String: Encodable]
     let userInfoAttributes: [String: Encodable]
 
+    /// Custom View timings (only available if `DM` is a RUM View model)
+    let customViewTimings: [String: Int64]?
```

Comment:
this comment doesn't sound like it's intended.?
IMO we can add `customViewTimings` via an `RUMView` extension
```swift
extension RUMView {
  func addCustomTimingAttribute(_ timing) { self.attributes["view.custom_timings.\(timing.name)"] = timing.value }
  // ...and if needed
  var customViewTimings { return attributes.filter { $0.startsWith("view.custom_timings") } }
}
```

Reply:
`RUMView` is created by the user to configure the auto-instrumentation and `UIKitRUMViewsPredicate`, I don't find it like a good place to store custom timings. Do you mean `RUMDataView: RUMDataModel`? That one in turn is auto-generated, so I didn't want to extend it with any custom functionality (even through extension in separate file).

As all other encoding logic is done in `RUMEventEncoder`, I found it quite natural to put timings also in there. Having additional property for timings (`customViewTimings`) will come handy for data scrubbing, as it adds clear context to those values and we can make them (un-)editable as we want. If merged with `attributes`, we'd have to additionally filter them out as you suggest, so I prefer the easiest way for now.

Reply:
sorry yes, i meant `RUMDataView`
ideally this was supposed to be part of `RUMDataView` but it's not so due to limitations of JSON schema
i'd stay as close as possible to the ideal solution

but i'm resolving because we just follow android here 👌